File: cue-command/src/naming.js
const crypto = require('crypto');

const C = 'bcdfghjklmnpqrstvwxz';
const V = 'aeiou';
const CODA = ['', 'n', 'r', 'l', 's', 'm', 'nd', 'st', 'rk', 'ld'];

function choice(strOrArr) {
  const n = strOrArr.length;
  const i = crypto.randomInt(0, n);
  return strOrArr[i];
}

function syllable() {
  return choice(C) + choice(V) + choice(CODA);
}

function pureName(minLen = 8, maxLen = 12) {
  for (let tries = 0; tries < 100; tries += 1) {
    const nSyl = choice([3, 4, 5]);
    let s = '';
    for (let i = 0; i < nSyl; i += 1) s += syllable();
    s = s.slice(0, maxLen);
    if (s.length >= minLen && s.length <= maxLen && /^[a-zA-Z]+$/.test(s)) {
      return s;
    }
  }
  throw new Error('Name generation failed; adjust syllable/length parameters');
}

function generateName() {
  return pureName().toLowerCase();
}

module.exports = { generateName };

File: cue-mcp/cuemcp/naming.py
"""Conversation name generator."""
import secrets


C = "bcdfghjklmnpqrstvwxz"
V = "aeiou"
CODA = ["", "n", "r", "l", "s", "m", "nd", "st", "rk", "ld"]


def syllable() -> str:
    return secrets.choice(C) + secrets.choice(V) + secrets.choice(CODA)


def pure_name(min_len: int = 8, max_len: int = 12) -> str:
    for _ in range(100):
        n_syl = secrets.choice([3, 4, 5])
        s = "".join(syllable() for _ in range(n_syl))
        s = s[:max_len]
        if min_len <= len(s) <= max_len and s.isalpha():
            return s
    raise RuntimeError("Name generation failed; adjust syllable/length parameters")


def generate_name() -> str:
    """Generate a human-friendly conversation name.

    Format: pronounceable lowercase letters only
    Examples: tavilron, nemosand
    """
    return pure_name().lower()
